Riot/ViewController/DirectoryServerPickerViewController.m

```diff
@@ -176,7 +176,7 @@ - (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath
 //        }
     }
 
-    [self dismissViewControllerAnimated:YES completion:nil];
+    [self.navigationController dismissViewControllerAnimated:YES completion:nil];
```

Comment:
You should use here  `withdrawViewControllerAnimated:completion` defined in the `MXKViewControllerHandling` protocol.